### ISSUE TITLE
Easily parallelizable reshape

### ISSUE THREAD

**Issue by HEmile** [HEmile/storchastic#22]
Reshape is quite difficult to make parallelizable: `logits = logits.reshape(logits.shape[:-1] + (20, 10))` works though. Maybe make it possible to use `logits.reshape[..., 20, 10]`?
